src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py

```diff
@@ -572,7 +572,7 @@ class XLMRobertaXLPreTrainedModel(PreTrainedModel):
 
     config_class = XLMRobertaXLConfig
     base_model_prefix = "roberta"
-    _no_split_modules = ["XLMRobertaXLEmbeddings", "XLMRobertaXLSelfAttention"]
+    _no_split_modules = ["XLMRobertaXLEmbeddings", "XLMRobertaXLLayer"]
```

Comment:
if you can leave a comment (in the PR page is ok) why this fix the issue, that would be very nice as knowledge sharing ❤️ .

Reply:
This is needed because there are a lot of residual connections at the level of the layer. So we need to make sure that all the ops that are happening inside of the layer stay on the same device. Otherwise, we would have to do manually move the tensors(.to(device) and this is not good. If you check other `_no_split_modules` with transformers architecture (llama, mistral ...), you will see that most of them have the layer class inside of it.

Reply:
Thanks a lot!